### PR TITLE
add dynamic audit policy controller

### PR DESCRIPTION
**Pull request by EmilyM1**
Removes static elements of audit policy and allows for dynamic creation with groups specified.

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -579,6 +578,13 @@ func prepareOauthAPIServerOperator(ctx context.Context, controllerContext *contr
 		operatorCtx.kubeClient.CoreV1(),
 		operatorCtx.kubeClient.CoreV1(),
 		operatorCtx.kubeInformersForNamespaces,
+	).WithAuditPolicyController(
+		"openshift-oauth-apiserver",
```
Comment (sttts) [inline pkg/operator/starter.go:582]: drop the `openshift-` prefix

**Comment by sttts** [comment]
The pod still tries to access `/var/run/configmaps/audit/default.yaml`. This must be `/var/run/configmaps/audit/policy.yaml` instead (in bindata).

File: pkg/operator/workload/sync_openshift_oauth_apiserver_test.go [hunk 2]
```diff
@@ -143,59 +143,6 @@ var emptyAPIServerArgsJSON = `
...
-			},
-		},
-
-		{
-			name: "default values are NOT assigned when provided",
-			spec: operatorv1.OperatorSpec{
-				ObservedConfig: runtime.RawExtension{Raw: []byte(withDefaultsProvidedAPIServerArgsJSON)},
-			},
-			expectedAPIServerArguments: map[string][]string{
-				"api-audiences":     {"https://now.something.different"},
```
Comment (sttts) [inline pkg/operator/workload/sync_openshift_oauth_apiserver_test.go:179]: can we preserve this part of the test about api-audiences?

Comment (sttts) [reply pkg/operator/workload/sync_openshift_oauth_apiserver_test.go:179]: just leave the test, but remove the lines with audit policy. That should work, shouldn't it?

Comment (EmilyM1) [reply pkg/operator/workload/sync_openshift_oauth_apiserver_test.go:179]: I can try it.

File: pkg/operator/workload/sync_openshift_oauth_apiserver_test.go [hunk 3]
```diff
@@ -160,16 +160,6 @@ func TestGetStructuredConfigWithDefaultValue(t *testing.T) {
 		spec                       operatorv1.OperatorSpec
 		expectedAPIServerArguments map[string][]string
 	}{
-		{
-			name: "default values are assigned when not provided",
-			spec: operatorv1.OperatorSpec{
-				ObservedConfig: runtime.RawExtension{Raw: []byte(emptyAPIServerArgsJSON)},
-			},
-			expectedAPIServerArguments: map[string][]string{
-				"audit-policy-file": {"/var/run/configmaps/audit/default.yaml"},
```
Comment (sttts) [inline pkg/operator/workload/sync_openshift_oauth_apiserver_test.go:169]: just remove this line only

Comment (sttts) [reply pkg/operator/workload/sync_openshift_oauth_apiserver_test.go:169]: And line 180 too

File: pkg/operator/workload/testdata/sync_ds_scenario_2.yaml [hunk 4]
```diff
@@ -47,23 +47,23 @@ spec:
...
-              --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 \
-              --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 \
-              --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 \
-              --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 \
-              --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305 \
-              --tls-cipher-suites=TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305 \
-              --tls-min-version=VersionTLS12 \
-              --v=2
+                --cors-allowed-origins='//127\.0\.0\.1(:|$)' \
+                --cors-allowed-origins='//localhost(:|$)' \
```
Comment (sttts) [inline pkg/operator/workload/testdata/sync_ds_scenario_2.yaml:58]: move left every but the first green line

File: pkg/operator/workload/testdata/sync_ds_scenario_3.yaml [hunk 5]
```diff
@@ -47,18 +47,19 @@ spec:
...
                 --shutdown-delay-duration=10s \
                 --tls-private-key-file=/var/run/secrets/serving-cert/tls.key \
                 --tls-cert-file=/var/run/secrets/serving-cert/tls.crt \
-                --audit-policy-file=/var/run/configmaps/audit/default.yaml \
-              --cors-allowed-origins='//127\.0\.0\.1(:|$)' \
-              --cors-allowed-origins='//localhost(:|$)' \
-              --tls-cipher-suites=TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305 \
-              --tls-min-version=VersionTLS13 \
-              --v=2
+                --audit-policy-file=/var/run/configmaps/audit/policy.yaml \
```
Comment (sttts) [inline pkg/operator/workload/testdata/sync_ds_scenario_3.yaml:57]: remove this line

File: pkg/operator/workload/testdata/sync_ds_scenario_3.yaml [hunk 6]
```diff
@@ -47,18 +47,19 @@ spec:
...
+                --cors-allowed-origins='//127\.0\.0\.1(:|$)' \
+                --cors-allowed-origins='//localhost(:|$)' \
```
Comment (sttts) [inline pkg/operator/workload/testdata/sync_ds_scenario_3.yaml:59]: move left from here on

**Comment by stlaz** [comment]
/hold
needs better commit management (and the tests to pass)

**Comment by EmilyM1** [comment]
@stlaz Put back in WIP till tests are passing and squashed.
Found in this PR we have quite a few previously existing yaml alignment problems, so flurry of commits from yesterday addresses those.
Will remove WIP and ping you when passes and resquashed if needed.

**Comment by stlaz** [comment]
/hold cancel
this seems ready

**Pull request merged**